beacon_node/eth2-libp2p/src/behaviour.rs

```diff
@@ -83,6 +83,10 @@ impl<TSubstream: AsyncRead + AsyncWrite> Behaviour<TSubstream> {
         &self.discovery
     }
 
+    pub fn discovery_mut(&mut self) -> &mut Discovery<TSubstream> {
```

Comment:
I don't think we should expose the entire discovery protocol. Perhaps in the future if we need it, but it seems like we only need to add an `add_enr()` function rather than get a mutable discovery

Reply:
Yup makes sense.